buf/registry/plugin/v1beta1/digest.proto

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.registry.plugin.v1beta1;
+
+import "buf/validate/validate.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/plugin/v1beta1";
+
+// A digest of a Commit's content.
+//
+// A digest represents all content for a single Commit.
+message Digest {
+  // The type of the Digest.
+  DigestType type = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The value of the Digest.
+  bytes value = 2 [(buf.validate.field).required = true];
+}
+
+// The type of Digest.
+enum DigestType {
+  DIGEST_TYPE_UNSPECIFIED = 0;
+  // The sha256 digest function.
+  DIGEST_TYPE_SHA256 = 1;
```

Comment:
Why is this here? This doesn't make sense per our conversation about digest functions.

Reply:
This hinged on the new API exposing docker-based protoc plugins. When fetching those plugins it'd be odd to exclude the digest.

If we remove docker-based protoc plugins from this API then we don't need this.

Reply:
I don't understand why you'd name the docker-based plugin digest type "sha256" - we have had conversations about how the hash algorithm is just one component of a digest algorithm. Names such as "docker_sha256", "docker1", or the like, would make more sense.

Reply:
Fair, this was poorly named based on our understanding, `docker_sha256` would have been more appropriate.

My initial thought was the element of least surprise since I'd wager when most users think about "docker digests" they are thinking of the sha26 digest.